code/modules/admin/topic.dm

```diff
@@ -734,9 +745,12 @@
 
 		//Create a list of unbanned jobs within job_list
 		var/list/notbannedlist = list()
+		var/list/bannedlist = list()
 		for(var/job in job_list)
 			if(!jobban_isbanned(M, job))
 				notbannedlist += job
+			else
+				bannedlist |= job
```

Comment:
This whole thing feels like it should just be tracked by the jobs subsystem honestly

Reply:
Honestly, this whole thing is a fucking mess lmao.
2k lines all in a single topic? Actually vile. But I feel like a re-work of this whole thing would likely fall out of scope of this PR. I'm not _too_ concerned as this only runs on-demand, when someone clicks ban/unban